rococo/repositories/mongodb/mongodb_repository.py

```diff
@@ -0,0 +1,145 @@
+"""MongoDbRepository class"""
+
+from dataclasses import fields
+from datetime import datetime
+import json
+from typing import Dict, List, Type
+from uuid import UUID
+import uuid
+from pymongo import ReturnDocument
+
+from rococo.data import MongoDBAdapter
+from rococo.messaging import MessageAdapter
+from rococo.models.versioned_model import VersionedModel
+from rococo.repositories import BaseRepository
+
+
+def get_uuid4_hex():
+    return uuid.uuid4().hex
+
+class MongoDbRepository(BaseRepository):
+    """MongoDbRepository class"""
+    def __init__(
+            self,
+            db_adapter: MongoDBAdapter,
+            model: Type[VersionedModel],
+            message_adapter: MessageAdapter,
+            queue_name: str,
+            user_id: UUID = None
+    ):
+        super().__init__(db_adapter, model, message_adapter, queue_name, user_id=user_id)
+        self.db = db_adapter.db
+
+
+    def _process_data_before_save(self, instance: VersionedModel):
+        """Method to convert VersionedModel instance to a data dictionary that can be inserted in MongoDB"""
+        super()._process_data_before_save(instance)
+        data = instance.as_dict(convert_datetime_to_iso_string=False, convert_uuids=True)
+        for field in fields(instance):
+            if data.get(field.name) is None:
+                continue
+
+            field_value = data[field.name]
+
+            if field.metadata.get('field_type') in ['entity_id', 'uuid']:
+                if isinstance(field_value, VersionedModel):
+                    field_value = str(field_value.entity_id).replace('-', '')
+                elif isinstance(field_value, dict):
+                    field_value = str(field_value.get('entity_id')).replace('-', '')
+                elif isinstance(field_value, str):
+                    field_value = field_value.replace('-', '')
+            if isinstance(field_value, UUID):
+                field_value = str(field_value).replace('-', '')
+            if isinstance(field_value, datetime):
+                field_value = field_value.strftime('%Y-%m-%d %H:%M:%S')
```

Comment:
The same code is in MySqlRepository. For now, it is okay. However, it's good to think where it should belong to avoid duplication. I imagine we will have more copies soon, and then we will have to move them.